lavalink/node.py

```diff
@@ -199,12 +293,9 @@ async def listener(self):
             if msg.type != aiohttp.WSMsgType.TEXT:
                 if msg.type is aiohttp.WSMsgType.ERROR:
                     log.debug("Ignoring exception in listener task", exc_info=msg.data)
-                elif msg.type in (
-                    aiohttp.WSMsgType.CLOSED,
-                    aiohttp.WSMsgType.CLOSING,
-                    aiohttp.WSMsgType.CLOSE,
-                ):
+                elif msg.type in self._closers:
                     log.debug("Listener closing: %s", msg.extra)
+                    await self._reconnect()
```

Comment:
There's a break after this `if` which results in `_reconnect()` getting awaited here first and later after leaving the `while` loop, a new task for `_reconnect()` gets spawned. If I'm getting this right, this means we end up reconnecting twice and possibly also end up with duplicated websocket connection and two listener() tasks as `connect()` does not have anything preventing it from connecting when it's already connected.